### PR TITLE
refactor: extract strings

### LINKED ISSUE #179: Refactor: extract string resources

### PR DESCRIPTION
**Pull request by aaisel**
# Description

This PR extracts hardcoded strings from the application code and moves them into the `strings.xml` file to improve maintainability. This change will simplify future updates to text content and make it easier to add support for other languages. 

Changes affect only strings that visible to the user, except for the following:
- `AddNote.kt` was not modified since it will be changed soon
- Toasts were not modified

Fixes #179 

# How Has This Been Tested?

No tests needed.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/Comments.kt [hunk 1]
```diff
@@ -172,11 +174,11 @@ fun CommentsSection(
                       if (comment.isUnedited()) {
                         "${comment.userName} : $formattedDate"
                       } else {
-                        "${comment.userName} edited: $formattedDate"
+                        stringResource(R.string.edited, comment.userName, formattedDate)
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/Comments.kt:177]: This is very well done, well played to figuring that string resources can take arguments

**Review by LawrenceLeitgib — CHANGES_REQUESTED** [review]
Very good job! This PR is incredibly well done. I have just one little thing that might need to be changed but except that we are perfect. Everything is well organize and make a lot of sense (the naming convention you choose for the string is very clear).

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -3,4 +3,73 @@
...
+    <string name="this_folder_is_empty">This folder is empty.</string>
+    <string name="first_name">First Name</string>
+    <string name="user_name">* User Name</string>
+    <string name="last_name">Last Name</string>
+    <string name="save">Save</string>
+    <string name="edit_profile">Edit Profile</string>
+    <string name="delete_account">Delete Account</string>
+    <string name="delete_account_prompt">Are you sure you want to delete your account?</string>
+    <string name="unsaved_changes">Unsaved changes</string>
+    <string name="go_back_without_saving_prompt">Are you sure you want to go back without saving changes?</string>
```
[the remark this answers is not in the document]
Comment (aaisel) [reply app/src/main/res/values/strings.xml]: Modified. Thank you!

**Review by LawrenceLeitgib — APPROVED** [review]
Nice correction I approve!

**Pull request merged**